README.md

```diff
@@ -1,492 +1,271 @@
-public class SomethingApiTests
+public class SomethingApiFixture : IDisposable
 {
-  [Fact]
-  public void EnsureSomethingApiHonoursPactWithConsumer()
-  {
-    //Arrange
-    const string serviceUri = "http://localhost:9222";
-    var config = new PactVerifierConfig
-    {
-        Outputters = new List<IOutput> //NOTE: We default to using a ConsoleOutput, however xUnit 2 does not capture the console output, so a custom outputter is required.
-        {
-            new XUnitOutput(_output)
-        },
-        CustomHeaders = new Dictionary<string, string>{{"Authorization", "Basic VGVzdA=="}}, //This allows the user to set request headers that will be sent with every request the verifier sends to the provider
-        Verbose = true //Output verbose verification logs to the test output
-    };
+    private readonly IHost server;
+    public Uri ServerUri { get; }
 
-    using (WebApp.Start<TestStartup>(serviceUri))
+    public SomethingApiFixture()
     {
-        //Act / Assert
-        IPactVerifier pactVerifier = new PactVerifier(config);
-
-        var pactUriOptions = new PactUriOptions()
-            .SetBasicAuthentication("someuser", "somepassword") // you can specify basic auth details
-            //or
-            .SetBearerAuthentication("sometoken") // Or a bearer token
-            //and / or
-            .SetSslCaFilePath("path/to/your/ca.crt") //if you fetch your pact in https and the server certificate authorities are not in the default ca-bundle.crt
-            //and / or
-            .SetHttpProxy("http://my-http-proxy:8080"); //if you need to go through a proxy to reach the server hosting the pact  
-
-        pactVerifier
-            .ProviderState($"{serviceUri}/provider-states")
-            .ServiceProvider("Something API", serviceUri)
-            .HonoursPactWith("Consumer")
-            .PactUri("..\\..\\..\\Consumer.Tests\\pacts\\consumer-something_api.json")
-            //or
-            .PactUri("http://pact-broker/pacts/provider/Something%20Api/consumer/Consumer/latest") //You can specify a http or https uri
-            //or
-            .PactUri("http://pact-broker/pacts/provider/Something%20Api/consumer/Consumer/latest", pactUriOptions) //With options decribed above
-            //or (if you're using the Pact Broker, you can use the various different features, including pending pacts)
-            .PactBroker("http://pact-broker", uriOptions: pactUriOptions, enablePending: true, consumerVersionTags: new   List<string> { "master" }, providerVersionTags: new List<string> { "master" }, consumerVersionSelectors: new List<VersionTagSelector> { new VersionTagSelector("master") })
-            .Verify();
+        ServerUri = new Uri("http://localhost:9222");
+        server = Host.CreateDefaultBuilder()
+                    .ConfigureWebHostDefaults(webBuilder =>
+                    {
+                        webBuilder.UseUrls(ServerUri.ToString());
+                        webBuilder.UseStartup<TestStartup>();

```

Comment:
I'd put a comment against this section explaining that you are expected to start your own web host that serves your real API., and this is a simple example of how to do that.

People frequently seem to ask for in-memory `TestHost` support (from the `Microsoft.AspNetCore.TestHost` package) and we simply can't support that.. Something like:

------

In order for Pact to call your API with the expectations from the contract, your API must be running on a real locally-accessible port using a real `Host`. This simple example shows how you can do that.

We are frequently asked whether a `TestHost` could be used instead (from the `Microsoft.AspNetCore.TestHost` package), but this isn't possible because the core Pact library is not running in .Net and therefore can't use an in-memory `HttpClient`.